### PR TITLE
LCDproc: Add a link status screen for each interface. Issue #8198

### PR DESCRIPTION
**Pull request by vktg**
Redmine Issue: https://redmine.pfsense.org/issues/8198
Ready for review

The link screen is similar to the 'Interfaces' dashboard widget, displaying ipv4, ipv6, mac and link status for each interface.

This is fixed/resolved copy of the original PR https://github.com/pfsense/FreeBSD-ports/pull/377 by @c-schwamborn

### CODE REVIEW

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php [hunk 1]
```diff
@@ -474,6 +474,51 @@ function outputled_gateway() {
...
+
+		// get the interface link infos
+		$ifinfo = get_interface_info($ifdescr);
+
+		$entry = array();
+		$entry['name'] = $ifname;
+		$entry['mac'] = $ifinfo['macaddr'];
+
+		if ($ifinfo['status'] == "up" ||
+		    $ifinfo['status'] == "associated") {
```
Comment (jim-p) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php]: Each test should be in parenthesis

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php [hunk 2]
```diff
@@ -474,6 +474,51 @@ function outputled_gateway() {
...
+
+			$entry['status'] = "up";
+		} else {
+			$entry['status'] = "down";
+		}
+
+		if ($ifinfo['pppoelink'] == "up" ||
+		    $ifinfo['pptplink']  == "up" ||
+		    $ifinfo['l2tplink']  == "up") {
```
Comment (jim-p) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php]: Each test should be in parenthesis

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php [hunk 3]
```diff
@@ -1181,6 +1251,37 @@ function loop_status($lcd) {
 					}
 					$updateSummary = false;
 					break;
+				case "scr_interfaces_link":
+					// We only want build_interface_link_list() to be
+					// called once per loop, and only if it's needed
+					if ($ifLinkList == null)
+						$ifLinkList = build_interface_link_list();
```
Comment (jim-p) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php]: The body of this `if` needs braces around it.

Comment (vktg) [reply sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php]: all fixed

**Pull request merged**